app.go

```diff
@@ -521,13 +414,13 @@ func Error(err error, code string, metadata ...map[string]string) *errors.Error
 }
 
 // GetSessionFromCtx retrieves a session from a given context
-func GetSessionFromCtx(ctx context.Context) *session.Session {
+func (app *App) GetSessionFromCtx(ctx context.Context) session.Session {
```

Comment:
Does this need to be a method? It does not use the receiver.

Reply:
That's a good question. It indeed doesn't need to be in practice. On the other hand, this is required to properly mock it. You can check in the refactored tests that I'm doing that.